scipy/stats/tests/test_morestats.py

```diff
@@ -2570,18 +2554,23 @@ def test_nan_propagate(self, test_func):
                                 1: 0.005545914017677123}),
                               (stats.circstd,
                                {None: np.nan, 0: 4.11093193, 1: 6.04265394})])
-    def test_nan_propagate_array(self, test_func, expected):
-        x = np.array([[355, 5, 2, 359, 10, 350, 1],
-                      [351, 7, 4, 352, 9, 349, np.nan],
-                      [1, np.nan, np.nan, np.nan, np.nan, np.nan, np.nan]])
+    def test_nan_propagate_array(self, test_func, expected, xp):
+        x = xp.asarray([[355, 5, 2, 359, 10, 350, 1],
+                        [351, 7, 4, 352, 9, 349, np.nan],
+                        [1, np.nan, np.nan, np.nan, np.nan, np.nan, np.nan]])
         for axis in expected.keys():
             out = test_func(x, high=360, axis=axis)
             if axis is None:
-                assert_(np.isnan(out))
+                xp_assert_equal(out, xp.asarray(xp.nan))
             else:
-                assert_allclose(out[0], expected[axis], rtol=1e-7)
-                assert_(np.isnan(out[1:]).all())
-
+                rtol = 2e-5 if is_torch(xp) else 1e-7
+                xp_assert_close(out[0], xp.asarray(expected[axis]), rtol=rtol)
+                xp_assert_equal(out[1:], xp.full_like(out[1:], xp.nan))
+
+    @pytest.mark.skip_xp_backends(np_only=True,
+                                  reasons=["Only NumPy has nan_policy='omit' for now"])
+    @pytest.mark.usefixtures("skip_xp_backends")
+    @array_api_compatible
```

Comment:
I know it's a little sloppy to mark these `@array_api_compatible` and then not actually use `xp` in the test. Here's what I'm trying to do:
- I meant to remove all the individual `array_api_compatible` decorators and decorate the class.
- I only want this test to run with NumPy backend, if at all.
- I don't want to convert this test to be array API compatible because it's a waste of time. `nan_policy` is implemented for these functions using the `_axis_nan_policy` decorator, which already performs much stronger property-based tests.

How about I just split these tests as they are into a separate class that is not decorated with `array_api_compatible`?